iotileship/iotile/ship/resources/filesystem_manager.py

```diff
@@ -0,0 +1,120 @@
+''' A Resource for operating on a Filesystem
+
+    Can work on either a directory or a unmounted block device
+
+    Important: mounting and unmounting of a block device requires the udisksctl utility, which is included with
+    most distributions of Ubuntu
+
+    All Steps using this Resource will use self.root as the root directory to find other files
+
+    Root directory is a pathlib.Path
+
+'''
+
+import subprocess
+import pathlib
+import sys
+
+from iotile.core.utilities.schema_verify import DictionaryVerifier
+from iotile.core.utilities.schema_verify import StringVerifier
+from iotile.core.exceptions import ArgumentError
+from .shared_resource import SharedResource
+
+RESOURCE_ARG_SCHEMA = DictionaryVerifier(desc="filesystem_manager arguments")
+RESOURCE_ARG_SCHEMA.add_required("res_path", StringVerifier("path to resource to work with. "
+                                                            "Specify file:// for a directory or block:// for an unmounted disk partition"))
+
+
+class FilesystemManagerResource(SharedResource):
+    '''
```

Comment:
Style guide has a multi-line comment block with text on the same line. Essentially, if the comment all fits on one line, you end up with `'''this is a comment'''`, but on multiline you end up with
```
'''This is a comment
That contains multiple lines
And is a haiku
'''
```